tests/application_development/code_relocation/testcase.yaml

```diff
@@ -0,0 +1,19 @@
+common:
+  arch_allow: arm
+  tags: linker
+tests:
+  tests.application_development.code_relocation:
+    filter: not CONFIG_CPU_HAS_NXP_MPU and dt_chosen_enabled("zephyr,itcm") and dt_chosen_enabled("zephyr,sram")
```

Comment:
is `dt_chosen_enabled("zephyr,sram")` really needed?

Reply:
> Besides something trivial to fix the problem is that this doesn't look like a test but it keeps being mainly a sample. To be considered a test we have to "test" something, making sure that variables are actually relocated and such for example.
> 
> Please, convert this to a real test, checking things at every step. Using `printk` is not useful at all in tests.

I see, let me try to convert it.